mlflow/utils/uri.py

```diff
@@ -129,3 +129,7 @@ def _join_posixpaths_and_append_absolute_suffixes(prefix_path, suffix_path):
     # joined path
     suffix_path = suffix_path.lstrip(posixpath.sep)
     return posixpath.join(prefix_path, suffix_path)
+
+
+def is_artifact_acled_uri(artifact_uri):
```

Comment:
In the same vein as the comment above about semantically equivalent URIs (e.g., `dbfs:///databricks/mlflow-tracking////EXP_ID/RUN_ID/...`), this check will fail. Can we do the following instead?:
1. Parse the URI
2. Extract the scheme and verify that it's `dbfs`
3. Extract the path, normalize it via `posixpath.normpath()`
4. Verify that the normalized path has the expected prefix

It would also be great to unit test this method against various semantically equivalent URIs.

Reply:
Done, however, I do not think step 2 is necessary since https://livegrep.dev.databricks.com/view/mlflow/mlflow/mlflow/store/artifact/artifact_repository_registry.py#L63 will only call `dbfs_artifact_repo_factory` when the scheme is dbfs.

Reply:
@arjundc-db I think we're relying on an assumption about exactly where this function is going to be called. Future callers of this function are not guaranteed to perform `dbfs` scheme validation beforehand. Accordingly, we have a few options:

1. Perform the scheme validation in this method
2. Do not perform the scheme validation in this method and clearly document (via the method docstring) that this method assumes it is being passed a `dbfs:/` URI and will not validate the scheme
3. If we don't foresee this method being called elsewhere, we can define it within the `dbfs_artifact_repo_factory()` method as a subroutine. This way, we can safely assume that we're already working with a `dbfs` URI (as documented by the `dbfs_artifact_repo_factory` docstring).

I'd vote for either option 1 or option 3!

Reply:
Yep, that makes sense. Using option 3.